pep-0458.txt

```diff
@@ -875,17 +859,22 @@ was susceptible to these attacks and how TUF could protect users against them
 
 __ https://mail.python.org/pipermail/distutils-sig/2013-September/022755.html
 
-Note that compromising *targets* or any delegated role (except for project
-targets metadata) does not immediately allow an attacker to serve malicious
+Note that compromising *targets* or *bins*
+does not immediately allow an attacker to serve malicious
 updates.  The attacker must also compromise the *timestamp* and *snapshot*
 roles, which are both online and therefore more likely to be compromised.
 This means that, in order to launch any attack, one must not only be able to
 act as a man-in-the-middle, but also compromise the *timestamp* key (or
 compromise the *root* keys and sign a new *timestamp* key).  To launch any
 attack other than a freeze attack, one must also compromise the *snapshot* key.
+In practice, this PEP recommends storing the *snapshot*, *timestamp*, and 
+*bin-n* keys together, or even using the same key for all of these roles.
+Because of this, the attacker only needs to compromise this single server to
+perform any of the attacks listed above. However, the offline *root* key will
+allow the repository to recover from an attack by revoking the online key(s).
 
 Finally, a compromise of the PyPI infrastructure MAY introduce malicious
-updates to *bins* projects because the keys for these roles are online.  The
+updates to *bin-n* projects because the keys for these roles are online.  The
```

Comment:
It feels like the leading two sentences in this paragraph pretty much repeat what has been said at the end of the previous paragraph. I suggest to remove them. 

To bridge from online key compromise to PEP 480. I suggest to rephrase the remainder of this paragraph to something like:

> The maximum security model shows how TUF mitigates online key compromises by introducing additional roles for end-to-signing. Details about how to generate developer keys and singing upload distributions are provided in PEP 480 [26]_.